modules/mcuboot/CMakeLists.txt

```diff
@@ -125,40 +125,51 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
   set(app_sign_depends
     $<IF:${sign_merged},${merged_hex_file_depends},zephyr_final>)
 
-  if (NOT DEFINED CONFIG_BOOT_SIGNATURE_KEY_FILE)
-    include(${CMAKE_BINARY_DIR}/mcuboot/shared_vars.cmake)
-    set(CONFIG_BOOT_SIGNATURE_KEY_FILE ${mcuboot_SIGNATURE_KEY_FILE})
-  endif ()
-
-  foreach (filepath ${mcuboot_CONF_FILE})
-    file(STRINGS ${filepath} mcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE
-         REGEX "^CONFIG_BOOT_SIGNATURE_KEY_FILE=")
-    if (mcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE)
-      get_filename_component(mcuboot_CONF_DIR ${filepath} DIRECTORY)
+  if(CONFIG_SIGN_IMAGES)
+    if (NOT CONFIG_MCUBOOT_BUILD_STRATEGY_FROM_SOURCE AND
+        NOT CONFIG_BOOT_SIGNATURE_KEY_FILE)
+      message(FATAL_ERROR
+        "Unable to sign the images since no signing key is specified. "
+        "When MCUboot is not built from source, it required that the signing "
+        "key is set explicitly through 'CONFIG_BOOT_SIGNATURE_KEY_FILE'")
```

Comment:
How about:
```suggestion
        "'CONFIG_BOOT_SIGNATURE_KEY_FILE' is not set."
        " Set 'CONFIG_BOOT_SIGNATURE_KEY_FILE=/path/to/<keyfile>.pem' or disable signing with 'CONFIG_SIGN_IMAGES=n' in 'prj.conf'"
```

in order to tell the user that signing can also be disabled.
It could be a user not having the signing key, and thus spend time on trying to figure out how or which key to use, not knowing that signing of app image can be disabled.

Maybe also check for the existence of the signing key, to fail early with good error message (I can't see that is done anywhere now)